examples/examples/runtime_call_enum.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! In some cases we are interested only in the `RuntimeCall` enum (or more generally, only in some
+//! runtime types). We can ask `subxt` to generate only runtime types by passing a corresponding
+//! flag.
+//!
+//! Here we present how to correctly create `Block` type for Polkadot chain.
+
+use sp_core::H256;
+use sp_runtime::{
+    generic,
+    traits::{
+        BlakeTwo256,
+        Block as _,
+        Header as _,
+    },
+    Digest,
+};
+use subxt::PolkadotConfig;
+
+#[subxt::subxt(
+    runtime_metadata_path = "../artifacts/polkadot_metadata.scale",
+    derive_for_all_types = "Clone, PartialEq, Eq",
+    runtime_types_only
+)]
+pub mod polkadot {}
+
+type RuntimeCall = polkadot::runtime_types::polkadot_runtime::RuntimeCall;
+
+type UncheckedExtrinsic = generic::UncheckedExtrinsic<
+    <PolkadotConfig as subxt::Config>::Address,
+    RuntimeCall,
+    <PolkadotConfig as subxt::Config>::Signature,
+    // Usually we are not interested in `SignedExtra`.
+    (),
+>;
+
+type Header = generic::Header<u32, BlakeTwo256>;
+type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    let polkadot_header = Header::new(
+        41,
+        H256::default(),
+        H256::default(),
+        H256::default(),
+        Digest::default(),
+    );
+
+    let polkadot_block = Block::new(polkadot_header, vec![]);
+
+    println!("{polkadot_block:?}");
+
+    Ok(())
+}
```

Comment:
This part of the example doesn't reallyhave anything to do with the "runtime types only" flag or making use of the `RuntimeCall` generated type?

Maybe just some code that encodes some UncheckedExtrinsic which is using the generated RuntimeCall would make more sense as an example? And Perhaps I'd rename it to `runtime_types_only.rs` but I'm not so wedded to that!

Reply:
Actually this example is my very motivation for the `runtime_types_only` flag :) I am using it in the alias `type RuntimeCall = polkadot::runtime_types::polkadot_runtime::RuntimeCall;`, which is required for specifying `UncheckedExtrinsic` type and then `Block` type. Usually, these types are created in runtime, where `RuntimeCall` enum is available from `construct_runtime!` macro. However, there are some usecases, where we are outside runtime, but still need to have `Block` type defined explicitly.

Nevertheless, if you prefer, I can change this example.

Reply:
I've renamed the example and added a commented-out snippet explaining what we lose when using `runtime_types_only` flag